x/programs/cmd/simulator/src/lib.rs

```diff
@@ -190,39 +221,10 @@ impl Client {
     /// # Errors
     ///
     /// Returns an error if the if serialization or plan fails.
-    pub fn run<T>(&self, plan: &Plan) -> Result<Vec<T>, Box<dyn Error>>
-    where
-        T: serde::de::DeserializeOwned + serde::Serialize,
```

Comment:
This was a premature abstraction. We shouldn't make things generic when there is only one type that can be used.